packages/@aws-cdk/aws-apigatewayv2/lib/common/base.ts

```diff
@@ -36,6 +36,11 @@ export abstract class StageBase extends Resource implements IStage {
   public abstract readonly stageName: string;
   protected abstract readonly baseApi: IApi;
 
+  /**
+   * The created ApiMapping if domain mapping has been added
+   */
+  protected apiMapping?: ApiMapping
```

Comment:
Can we make this internal for now?

```suggestion
  /**
   * The created ApiMapping if domain mapping has been added
   * @internal
   */
  protected _apiMapping?: ApiMapping
```

This will hide the API from JSII giving us the flexibility to change the implementation without it being a breaking change.